### DISCUSSION TITLE
Add Delete And Edit Button in pc.table

### DISCUSSION THREAD

**Discussion by bobwatcherx** [reflex-dev/reflex#946] [Q&A]
is there an example of adding edit and delete button in pc.table .

**Comment by jpbaltazar** [comment]
I have done this with the rx.foreach and a custom function. This allows you to edit/delete an entity by calling it from the state and embedding a unique identifier for the item in the list through a lambda function call.

	class Entity(rx.Model, table=False):
	    name: str
		unique_id: int
	
	class State(rx.State):
		entity_list: list[Entity] = []
		
		def edit_entity(unique_id):
			print(f"Editing entity {unique_id}")
		
		def delete_entity(unique_id):
			print(f"Deleting entity {unique_id}")
	
	def display_row(entity):
		return rx.tr(
			rx.td(
				rx.text(entity.name)
			),
			rx.td(
				rx.hstack(
					rx.icon(
						tag="edit",
						on_click=lambda: State.edit_entity(entity.unique_id),
					),
					rx.icon(
						tag="delete",
						on_click=lambda: State.delete_entity(entity.unique_id),
					),
				),
			),
		)
	
	def index():
		return rx.vstack(
			rx.table(
				rx.thead(),
				rx.tbody(
					rx.foreach(
						State.entity_list,
						display_row,
					)
				)
			),
			width="50%",
			height="100vh",
		)
